cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftRequest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.draft;
+
+import io.cdap.cdap.api.artifact.ArtifactSummary;
+
+import java.util.Objects;
+
+/**
+ * Request to store a draft.
+ *
+ * @param <T> Type of config that this DraftRequest contains
+ */
+public class DraftRequest<T> {
+  private final String previousHash;
+  private final String name;
+  private final String description;
+  private final int revision;
+  private final ArtifactSummary artifact;
+  private T config;
+
+  public DraftRequest(T config, String previousHash, String name, String description, int revision,
```

Comment:
If you provide a Builder, the constructor should be private.

Also, from the builder implementation, both the config and the artifact can be `null`. Is it intended? If so, please annotate them as `@Nullable`. If not, please check and throw exception in the Builder.build method.

Reply:
Removed the builder and make the constructor public